TYPO3.Neos/Classes/TYPO3/Neos/Service/View/NodeView.php

```diff
@@ -237,7 +237,9 @@ public function collectParentNodeData(NodeInterface $rootNode, array $nodes)
                     $collectTreeNodeData($children, $childNode);
                 }
             }
-            $treeNodes[] = $self->collectTreeNodeData($node['node'], true, $children, $children !== array(), isset($node['matched']));
+            if (isset($node['node']) && $node['node'] instanceof NodeInterface) {
```

Comment:
this isn't really the right place for such a check since it shouldn't happen.. instead a check in line 210 is more appropriate?